### PR TITLE
Dragdrop image widget

### PR DESCRIPTION
**Pull request by alchemistgo87**

### CODE REVIEW

File: packages/@holaplexui-playground/pages/form.tsx [hunk 1]
```diff
@@ -1,67 +1,67 @@
...
     >
       <path
-        d="M9.70514 9.03858C9.27877 9.75749 8.45061 10.1358 7.62805 9.98746C6.80549 9.83911 6.16168 9.1953 6.01333 8.37274C5.86497 7.55018 6.2433 6.72202 6.96221 6.29565"
-        stroke="black"
-        stroke-linecap="round"
-        stroke-linejoin="round"
+        d='M9.70514 9.03858C9.27877 9.75749 8.45061 10.1358 7.62805 9.98746C6.80549 9.83911 6.16168 9.1953 6.01333 8.37274C5.86497 7.55018 6.2433 6.72202 6.96221 6.29565'
+        stroke='black'
+        stroke-linecap='round'
+        stroke-linejoin='round'
```
Comment (kespinola) [inline packages/@holaplexui-playground/pages/form.tsx]: This dashes aren't valid prop types in react can you find and replace them all to `storkeLinecap` and `strokeLinejon` this is something we need to do with every svg all the dash casing needs to be converted to camel.

File: packages/@holaplexui-playground/pages/form.tsx [hunk 2]
```diff
@@ -129,15 +131,17 @@ export default function App() {
...
             setMultiValue(v);
           }}
           value={multiValue}
         >
-          <Form.Select.Button icon={<Icon.Sol />} placeholder="e.g. Solana">
-            {singleValue?.name}
+          <Form.Select.Button icon={<Icon.Sol />} placeholder='e.g. Solana'>
+            {multiValue && multiValue?.length > 0
+              ? multiValue[0].name
+              : undefined}
```
Comment (kespinola) [inline packages/@holaplexui-playground/pages/form.tsx:161]: Lets fix the children prop of this if its undefined then show placeholder.

File: packages/@holaplexui-react/src/components/Form.tsx [hunk 3]
```diff
@@ -207,26 +201,26 @@ function FormSelectButton({
...
-        {
-          'pl-12': icon,
-        },
-      )}
+      className={clsx('w-full inline-block', 'form-select-button', {
+        'pl-12': icon,
+      })}
     >
-      {children ? <span>{children}</span> : <span className="form-select-button-placeholder">{placeholder}</span>}
+      {children ? (
```
Comment (kespinola) [inline packages/@holaplexui-react/src/components/Form.tsx:210]: ok it already handles the children being undefined just adjust the prop types then to allow for undefined.

File: packages/@holaplexui-react/src/components/Form.tsx [hunk 4]
```diff
@@ -309,3 +303,50 @@ const FormCheckbox = forwardRef(function FormCheckbox(
...
+      <input {...getInputProps()} />
+      <div
+        className={clsx('dragdropimage', {
+          'dragdropimage-drag-active': isDragActive,
+          'dragdropimage-empty': !file,
+        })}
+      >
+        {file ? (
+          <div className="dragdropimage-image-container">
+            <img src={URL.createObjectURL(file)} alt={file.name} className="dragdropimage-image" />
```
Comment (kespinola) [inline packages/@holaplexui-react/src/components/Form.tsx]: The value should just be an imageurl string not a file can make the the alt the image url as well.

File: packages/@holaplexui-react/src/components/Form.tsx [hunk 5]
```diff
@@ -309,3 +303,50 @@ const FormCheckbox = forwardRef(function FormCheckbox(
 });
 
 Form.Checkbox = FormCheckbox;
+
+interface DragDropImageProps {
+  onChange: (file: File) => void;
+  value: File | null;
```
Comment (kespinola) [inline packages/@holaplexui-react/src/components/Form.tsx]:
```suggestion
  value: string | undefined;
```

File: packages/@holaplexui-react/src/components/Form.tsx [hunk 6]
```diff
@@ -309,3 +303,50 @@ const FormCheckbox = forwardRef(function FormCheckbox(
...
+    accept: {
+      'image/png': ['.png'],
+      'image/jpeg': ['.jpeg', '.jpg'],
+      'image/svg': ['.svg'],
+    },
+    maxSize: 2 * 1024 * 1024,
+    onDrop: (acceptedFiles) => {
+      if (acceptedFiles.length > 0) {
+        const file = acceptedFiles[0];
+        setFile(file);
```
Comment (kespinola) [inline packages/@holaplexui-react/src/components/Form.tsx]: Should not use internal state
```suggestion
```

**Review by kespinola — CHANGES_REQUESTED** [review]

File: packages/@holaplexui-playground/pages/form.tsx (hunk 2 above)
Comment (alchemistgo87) [reply packages/@holaplexui-playground/pages/form.tsx:161]: Yes, it already shows the placeholder if children are undefined.

File: packages/@holaplexui-react/src/components/Form.tsx [hunk 7]
```diff
@@ -309,3 +305,38 @@ const FormCheckbox = forwardRef(function FormCheckbox(
...
+    accept: {
+      'image/png': ['.png'],
+      'image/jpeg': ['.jpeg', '.jpg'],
+      'image/svg': ['.svg'],
+    },
+    maxSize: 2 * 1024 * 1024,
+    onDrop: (acceptedFiles) => {
+      if (acceptedFiles.length > 0) {
+        const file = acceptedFiles[0];
+        onChange(URL.createObjectURL(file));
```
Comment (kespinola) [inline packages/@holaplexui-react/src/components/Form.tsx]: The onChange should still be a file because it needs to be uploaded.

**Review by kespinola — CHANGES_REQUESTED** [review]

File: packages/@holaplexui-playground/hooks/useUpload.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+interface UploadContext {
+  getInputProps: <T extends DropzoneInputProps>(props?: T | undefined) => T;
+  getRootProps: <T extends DropzoneRootProps>(props?: T | undefined) => T;
+  isDragActive: boolean;
+  droppedFiles: File[] | null;
+}
+
+export default function useUpload(): UploadContext {
+  const [droppedFiles, setDroppedFiles] = useState<File[] | null>(null);
```
Comment (kespinola) [inline packages/@holaplexui-playground/hooks/useUpload.ts]: You don't need state looking at the docs you can pass an onChange to the getInputProps({ onChange }) don't pass through the hook. Pass to the DragDrop component.

File: packages/@holaplexui-react/src/components/Form.tsx [hunk 9]
```diff
@@ -306,37 +302,36 @@ const FormCheckbox = forwardRef(function FormCheckbox(
...
+  getInputProps,
+  getRootProps,
+  isDragActive,
+  className,
+  children,
+}: DragDropProps) => {
   return (
-    <div className={clsx('relative', className)}>
+    <div {...getRootProps()} className={clsx('relative', className)}>
       <input {...getInputProps()} />
```
Comment (kespinola) [inline packages/@holaplexui-react/src/components/Form.tsx]: Component should take onChange as a prop and pass to getInputProps({ onChange })

**Review by kespinola — CHANGES_REQUESTED** [review]

**Review by kespinola — APPROVED** [review]
Looks good. Update the package version and publish!

**Pull request merged**